taskflow.go

```diff
@@ -46,7 +44,56 @@ type RegisteredTask struct {
 func New() *Taskflow {
 	return &Taskflow{
 		Output: DefaultOutput,
-		Params: Params{},
+	}
+}
+
+// RegisterValueParam registers a generic parameter that is defined by the calling code.
+// Use this variant in case the primitive-specific implementations cannot cover the parameter.
+func (f *Taskflow) RegisterValueParam(newValue func() Value, info ParameterInfo) ValueParam {
```

Comment:
I think we can improve API for this method but let's tackle it in separate PR

Reply:
It would be great just to add an example showing how to use a custom parameter type. I suggest renaming `examples/string-param/main.go` to `examples/parameters/main.go` and add an example the for custom parameter type.

The current API is great 👍

Reply:
Documentation and example extended.